site/source/docs/porting/pthreads.rst

```diff
@@ -43,17 +43,67 @@ but is unrelated. That flag does not use pthreads or SharedArrayBuffer, and
 instead uses a plain Web Worker to run your main program (and postMessage to
 proxy messages back and forth).
 
+Blocking on the main browser thread
+===================================
+
+The Web API for atomics does not allow blocking on the main thread
+(specifically, ``Atomics.wait`` doesn't work there). Such blocking is
+necessary in APIs like ``pthread_join`` and anything that uses a futex wait
+under the hood, like ``usleep()``, ``emscripten_futex_wait()``, or
+``pthread_mutex_lock()``. To make them work, we use a busy-wait on the main
+browser thread, which can make the browser tab unresponsive, and also wastes
+power. (On a pthread, this isn't a problem as it runs in a Web Worker, where
+we don't need to busy-wait.)
+
+Busy-waiting on the main browser thread in general will work despite the
+downsides just mentioned, for things like waiting on a mutex.
+However, things like ``pthread_join`` and ``pthread_cond_wait``
+can block for long periods of time, and if that
+happens on the main browser thread, and while other threads expect it to
```

Comment:
I would just remove "while other threads expect it to respond' because even if other threads don't expect it, the user expects responsive UI.

Reply:
Oh wait I see that this is actually intended to describe the deadlock case.
So I guess if the user code on other threads are expecting a response, the main thread wouldn't be joining/waiting.

So maybe there are 2 things we are trying to say here. One is to highlight the spinning behavior, which is important to be aware of even in the absence of deadlocks. And the other is the possibility of deadlocks that wouldn't have existed in native code because of proxying.

I see you moved the discussion of proxying up from its own section to here, which makes sense because it's related.
I wonder if it would make more sense to put that in its own section first, then folllowed by the section on blocking, which would just refer to the previous section. I guess both proxying and blocking could be subsections under "special considerations" (or maybe they are important enough to get their own top-level section, followed by "other special considerations")